src/http/server.clj

```diff
@@ -293,6 +294,24 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn pw-prompt []
+  (str (do (println "Password: ") (.readPassword (System/console)))))
+
+(defn maybe-inject-auth
+  [auth handler]
+  (fn [req]
+    (handler (if auth
+               (assoc-in req [:headers "authorization"] auth)
+               req))))
+
+(defn parse-auth [auth]
+  "make sure pw is present, if not prompt for it"
+  (let [[user pw] (clojure.string/split auth #":")]
```

Comment:
`clojure.string` is already imported as `str`

Reply:
What if `split` returns > 2 elements?

Reply:
Oops - should have noticed the import already existed - and either way probably should have imported myself if it hadn't rather than using fully qualified name - that was leftover from repl experimentation.  

For your second point - I did consider this, but it would only happen if ":" is present in either username or password.  I didn't know if we wanted to go down the path of password validation, restricting PW's to a certain set of rules.  I did a little research and it seems that typically basic auth typically does not allow ":" to be present for username, but potentially does in password.  

How would you like this to work?  What about:

- Restrict username to alphanumeric only, allow all passwords
- Restrict username to any non ":" character, allow all passwords e.g. `(re-find #"([^:]*):(.*)" "user1:pa:ss:word2" )` 

Do we need to notify users of this restriction somehow, either in the long description of the CLI flag or elsewhere?  If a non-conforming username is submitted do we want interactive feedback via a prompt to tell them to choose a conforming pw?  Throw an error?  Just pass the invalid credentials along and let the server handle it?